### PR TITLE
Qualys_Add_New_Commands

### PR DESCRIPTION
**Pull request by DaniSalcedoGFT**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
We added five new commands to Qualys due to new requirements of our project that needed more functionality of existing
functionality.
This includes functionality regarding of qualys-update-unix-record and new functionality in order to map ESXi actives against the VCenters

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by adi88d — CHANGES_REQUESTED** [review]
Hi @DaniSalcedoGFT and thank you for your contribution!
- Update release notes by using [demisto-sdk update-release-notes](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/update_release_notes/README.md) command.
- please format the `Qualysv2.yml` file and keep only the command you added.
- add the new commands to the `README.md` file by using [demisto-sdk generate-docs](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/generate_docs/README.md) command.
Please feel free to reach out to me here or on Slack.

**Comment by DaniSalcedoGFT** [comment]
Hi @adi88d,
I pushed some of the changes but when I try to add the commands to the README.md it gives me this error, can you help me?

Failed generating docs for Packs/qualys/Integrations/Qualysv2/Qualysv2.yml
Traceback (most recent call last):
  File "/home/disa/.local/lib/python3.8/site-packages/demisto_sdk/__main__.py", line 2269, in _generate_docs_for_file
    raise Exception(
Exception: File is not an Integration, Script, Playbook or a README.

**Comment by adi88d** [comment]
> Hi @adi88d, I pushed some of the changes but when I try to add the commands to the README.md it gives me this error, can you help me?
> 
> Failed generating docs for Packs/qualys/Integrations/Qualysv2/Qualysv2.yml Traceback (most recent call last): File "/home/disa/.local/lib/python3.8/site-packages/demisto_sdk/**main**.py", line 2269, in _generate_docs_for_file raise Exception( Exception: File is not an Integration, Script, Playbook or a README.

Use the command for the **Packs/qualys/Integrations/Qualysv2/Qualysv2.yml** file

**Comment by DaniSalcedoGFT** [comment]
Hi @adi88d, not sure if i'm doing anything wrong but im executing the command and it seems is not working

disa@GFT-RxNNKG5QtdM:~/danibbva/content$ demisto-sdk generate-docs -i Packs/qualys/Integrations/Qualysv2/Qualysv2.yml -e Packs/qualys/Integrations/Qualysv2/command_examples
[INFO] Log file location: /home/disa/danibbva/content/demisto_sdk_debug.log
[INFO] You are using demisto-sdk 1.24.0.
[INFO] however version 1.25.1 is available.
To update, run pip3 install --upgrade demisto-sdk
Failed generating docs for Packs/qualys/Integrations/Qualysv2/Qualysv2.yml
Traceback (most recent call last):
  File "/home/disa/.local/lib/python3.8/site-packages/demisto_sdk/__main__.py", line 2269, in _generate_docs_for_file
    raise Exception(
Exception: File is not an Integration, Script, Playbook or a README.

**Comment by adi88d** [comment]
Hi @DaniSalcedoGFT, I will take the PR from here
Thanks again.

**Comment by adi88d** [comment]
I want to have a zoom session with you and go over the PR
you can join the DFIR Slack Community via [this link](https://start.paloaltonetworks.com/join-our-slack-community), fill the form and follow the instructions and we can speak there
or send me email to adaud@paloaltonetworks.com

**Comment by DaniSalcedoGFT** [comment]
Hi @adi88d I have very limited time so it will be difficult to make a session to review the PR, you need some explanation about the new commands or something? Should I send you some screenshots or some text explaining the functionality of the commands?

File: Packs/qualys/Integrations/Qualysv2/Qualysv2.yml [hunk 1]
```diff
@@ -2718,7 +2718,49 @@ script:
     - contextPath: Qualys.Purge.ID
       description: IDs of the hosts queued for purging.
     description: Purge hosts in your account to remove the assessment data associated with them.
-  dockerimage: demisto/python3:3.10.13.80014
+  - name: qualys-update-vmware-record
+    arguments:
+    - name: ids
+      required: true
+      description: Record IDs to update. Specify record IDs and/or ID ranges Multiple entries are comma separated.
```
Comment (ShirleyDenkberg) [inline Packs/qualys/Integrations/Qualysv2/Qualysv2.yml:2725]:
```suggestion
      description: A comma-separated list of record IDs to update. Specify record IDs and/or ID ranges.
```

File: Packs/qualys/Integrations/Qualysv2/Qualysv2.yml [hunk 2]
```diff
@@ -2718,7 +2718,49 @@ script:
...
+    - name: add_ips
+      required: true
+      description: Add IPs and/or ranges to the IPs list for this record. Multiple IPs/ranges are comma separated.
```
Comment (ShirleyDenkberg) [inline Packs/qualys/Integrations/Qualysv2/Qualysv2.yml:2728]:
```suggestion
      description: A comma-separated list of IPs and/or ranges to add to the IPs list for this record.
```

File: Packs/qualys/Integrations/Qualysv2/Qualysv2.yml [hunk 3]
```diff
@@ -2718,7 +2718,49 @@ script:
...
+    description: Update Vmware records for authenticated scans of hosts running on Vmware.
+  - name: qualys-update-vcenter-record
+    arguments:
+    - name: ids
+      required: true
+      description: Record IDs to update. Specify record IDs and/or ID ranges Multiple entries are comma separated.
```
Comment (ShirleyDenkberg) [inline Packs/qualys/Integrations/Qualysv2/Qualysv2.yml:2734]:
```suggestion
      description: A comma-separated list of record IDs to update. Specify record IDs and/or ID ranges.
```

File: Packs/qualys/Integrations/Qualysv2/Qualysv2.yml [hunk 4]
```diff
@@ -2718,7 +2718,49 @@ script:
...
+    - name: add_ips
+      required: true
+      description: Add IPs and/or ranges to the IPs list for this record. Multiple IPs/ranges are comma separated.
```
Comment (ShirleyDenkberg) [inline Packs/qualys/Integrations/Qualysv2/Qualysv2.yml:2737]:
```suggestion
      description: A comma-separated list of IPs and/or ranges to add to the IPs list for this record.
```

File: Packs/qualys/Integrations/Qualysv2/README.md [hunk 5]
```diff
@@ -21911,3 +21911,155 @@ There is no context output for this command.
...
+#### Base Command
+
+`qualys-update-vmware-record`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| ids | Record IDs to update. Specify record IDs and/or ID ranges Multiple entries are comma separated. | Required | 
+| add_ips | Add IPs and/or ranges to the IPs list for this record. Multiple IPs/ranges are comma separated. | Required |
```
Comment (ShirleyDenkberg) [inline Packs/qualys/Integrations/Qualysv2/README.md:21929]:
```suggestion
| ids | A comma-separated list of record IDs to update. Specify record IDs and/or ID ranges. | Required | 
| add_ips | A comma-separated list of IPs and/or ranges to add to the IPs list for this record. | Required | 
```

File: Packs/qualys/Integrations/Qualysv2/README.md [hunk 6]
```diff
@@ -21911,3 +21911,155 @@ There is no context output for this command.
...
+#### Base Command
+
+`qualys-update-vcenter-record`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| ids | Record IDs to update. Specify record IDs and/or ID ranges Multiple entries are comma separated. | Required | 
+| add_ips | Add IPs and/or ranges to the IPs list for this record. Multiple IPs/ranges are comma separated. | Required |
```
Comment (ShirleyDenkberg) [inline Packs/qualys/Integrations/Qualysv2/README.md:21956]:
```suggestion
| ids | A comma-separated list of record IDs to update. Specify record IDs and/or ID ranges. | Required | 
| add_ips | A comma-separated list of IPs and/or ranges to add to the IPs list for this record. | Required | 
```

**Comment by ShirleyDenkberg** [comment]
@adi88d Doc review completed.

**Comment by adi88d** [comment]
@DaniSalcedoGFT 
Please sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=31917) before we can accept your contribution.

**Comment by DaniSalcedoGFT** [comment]
I signed the License, thanks!

**Comment by adi88d** [comment]
Hi @DaniSalcedoGFT your commit made by different github user (Dani) and this is why the 
**license/cla validation** is not working:
image

in order to sign the Contributor License Agreement you need to do that with the user Dani, the one that did the commits before.

if you can not sign with the user Dani please add the email that associated with Dani to your account, see [this doc](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/adding-an-email-address-to-your-github-account)
and then sign the[Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=31917) again.

if adding an email address isn't working then follow [this instructions](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user)

and then sign the[Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=31917) again.


Thanks again!!

**Review by adi88d — APPROVED** [review]

**Comment by adi88d** [comment]
> I signed again adding the email to this account, it seems that now is ok, can you check it?

Thanks @DaniSalcedoGFT yes, the cla validation passed now

**Comment by adi88d** [comment]
@DaniSalcedoGFT Your contribution is in the marketplace In the pack **Qualys** V2.0.11

**Comment by DaniSalcedoGFT** [comment]
Ok, thanks!

**Pull request merged**